src/test/scala/com/microsoft/hyperspace/index/DeltaLakeIntegrationTest.scala

```diff
@@ -310,6 +321,264 @@ class DeltaLakeIntegrationTest extends QueryTest with HyperspaceSuite {
+      import spark.implicits._
+      val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+      dfFromSample.write.format("delta").save(dataPath)
+
+      val timestampMap = mutable.Map[Long, String]()
+      timestampMap.put(0, getSparkFormattedTimestamps(System.currentTimeMillis).head)
+
+      val deltaDf = spark.read.format("delta").load(dataPath)
+      withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> "true") {
+        hyperspace.createIndex(deltaDf, IndexConfig("deltaIndex", Seq("clicks"), Seq("Query")))
+      }
+
+      withIndex("deltaIndex") {
+        def query(version: Option[Long] = None, asTimestamp: Boolean = false): DataFrame = {
+          val deltaDf = if (version.isDefined) {
+            if (asTimestamp) {
+              spark.read
+                .format("delta")
+                .option("timestampAsOf", timestampMap(version.get))
+                .load(dataPath)
+            } else {
+              spark.read.format("delta").option("versionAsOf", version.get).load(dataPath)
+            }
+          } else {
+            spark.read.format("delta").load(dataPath)
+          }
+          deltaDf.filter(deltaDf("clicks") <= 2000).select(deltaDf("query"))
+        }
+
+        def checkExpectedIndexUsed(
+            timeTravelVersion: Option[Long],
+            expectedIndexPathSubStr: String*): Unit = {
+          assert(
+            isIndexUsed(
+              query(timeTravelVersion, asTimestamp = false).queryExecution.optimizedPlan,
+              expectedIndexPathSubStr: _*),
+            s"timeTravelVer: $timeTravelVersion, expectedPaths: $expectedIndexPathSubStr)")
+          if (timeTravelVersion.isDefined) {
+            // Check time travel query with timestampAsOf instead of versionAsOf.
+            val df = spark.read.format("delta").load(dataPath)
+            val latestVer = getDeltaLakeTableVersion(df)
+            // Delta Lake throws an exception if the timestamp is larger than the latest commit.
+            if (latestVer != timeTravelVersion.get) {
+              assert(
+                isIndexUsed(
+                  query(timeTravelVersion, asTimestamp = true).queryExecution.optimizedPlan,
+                  expectedIndexPathSubStr: _*),
+                s"timeTravelVer: $timeTravelVersion, expectedPaths: $expectedIndexPathSubStr)")
+            }
+          }
+        }
+
+        checkExpectedIndexUsed(None, "deltaIndex/v__=0")
+
+        def appendData(doRefresh: Boolean): Unit = {
+          // Append data.
+          dfFromSample
+            .limit(3)
+            .write
+            .format("delta")
+            .mode("append")
+            .save(dataPath)
+
+          // Get the timestamp string for time travel query using "timestampAsOf" option.
+          val timestampStr = getSparkFormattedTimestamps(System.currentTimeMillis).head
+          // Sleep 1 second because the unit of commit timestamp is second.
+          Thread.sleep(1000)
```

Comment:
Avoid using sleep in ut as it will take too much time especially when this method is called multiple times.

To simulate time just take a global Time variable `currTime =  System.currentTimeMillis` and then keep on adding 1000 to it whenever you want to move time and then just use with `getSparkFormattedTimestamps` function

Reply:
I added this because delta lake internally records the time in **second** unit. So I forces to record different times for each version update, so that I could test the query with timestamp properly.

Reply:
I don't think so, I ran your test and used ` DeltaTable.forPath(spark, dataPath).history().show(10, false)` to view timestamp (2021-02-02 14:49:19.077). It used milliseconds and each op should have a readVersion ?

Reply:
You can use the timestamp in delta table to store read versions ?

Reply:
Seems Azure test framework needs sleep for some reason :D.. https://github.com/microsoft/hyperspace/pull/272/commits/f19614b5b32a284137d1823d3a73f06f024a9067
I might have added this after debugging / investigation.

Reply:
> I added this because delta lake internally records the time in **second** unit. So I forces to record different times for each version update, so that I could test the query with timestamp properly.

Maybe add this comment to the code? Reading the existing comment, I didn't know why we had to sleep.